### DISCUSSION TITLE
User pooling?

### DISCUSSION THREAD

**Discussion by vicary** [supabase/supabase#9771] [Questions] [answered]
Is it possible to create two groups of users, one internal and one public, with possible duplicated emails between the two but not within, in the same project?

**Comment by dshukertjr** [comment, marked as answer]
You could probably create two tables `public.users` and `public.internal_users` that both have foreign key relationship to `auth.users` table. You can hide or expose those two tables as you please using [row level security policy](https://supabase.com/docs/guides/auth/row-level-security)!

**Reply by vicary** [reply]
I can separate two user pools with this method, but is it possible to allow the same email address separately registering for internal and public endpoints?

I mean, if `public.users` and `public.internal_users` both references the same `auth.users`, it should not allow the same email address registering more than once.

**Reply by dshukertjr** [reply]
@vicary 
Ah, I see what you want to do. This might change in the long run, but at the moment I don't think it is possible to have two separate registered users with the same email address.

**Reply by vicary** [reply]
@dshukertjr That's unfortunate but appreciate the clear answer, thanks!
